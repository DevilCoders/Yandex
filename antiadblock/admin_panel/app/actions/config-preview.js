export const OPEN_CONFIG_PREVIEW = 'OPEN_CONFIG_PREVIEW';
export const CLOSE_CONFIG_PREVIEW = 'CLOSE_CONFIG_PREVIEW';

export function openConfigPreview(serviceId, id, configData) {
    return {
        type: OPEN_CONFIG_PREVIEW,
        serviceId,
        id,
        configData
    };
}

export function closeConfigPreview(id) {
    return {
        type: CLOSE_CONFIG_PREVIEW,
        id
    };
}
