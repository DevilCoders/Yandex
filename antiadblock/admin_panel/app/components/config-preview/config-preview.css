.config-preview__radio
{
    position: absolute;
    right: 40px;
    top: 15px;
    width: 140px;

    & .radio-button
    {
        width: 100%;
    }
}
