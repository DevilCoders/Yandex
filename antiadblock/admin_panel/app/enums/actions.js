export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
