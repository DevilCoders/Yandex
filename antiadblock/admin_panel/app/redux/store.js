import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
