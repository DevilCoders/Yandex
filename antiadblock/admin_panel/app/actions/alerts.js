import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
