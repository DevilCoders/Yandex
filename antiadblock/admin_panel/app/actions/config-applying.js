import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
