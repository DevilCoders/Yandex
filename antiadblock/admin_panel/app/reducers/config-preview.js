import * as configPreviewActions from 'app/actions/config-preview';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        id: null,
        serviceId: null,
        configData: null
    };

    state = state || initialState;

    switch (type) {
        case configPreviewActions.OPEN_CONFIG_PREVIEW: {
            return {
                visible: true,
                id: action.id,
                serviceId: action.serviceId,
                configData: action.configData
            };
        }
        case configPreviewActions.CLOSE_CONFIG_PREVIEW: {
            return initialState;
        }
        default: {
            return state;
        }
    }
}
