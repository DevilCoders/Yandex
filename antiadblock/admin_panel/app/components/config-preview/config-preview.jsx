import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';

import YAML from 'js-yaml';
import ConfigEditor from 'app/components/config-editor/config-editor';
import ConfigYamlEditor from '../config-yaml-editor/config-yaml-editor';
import Bem from 'app/components/bem/bem';
import Modal from 'lego-on-react/src/components/modal/modal.react';
import Button from 'lego-on-react/src/components/button/button.react';
import RadioButton from 'lego-on-react/src/components/radio-button/radio-button.react';
import ModalClose from 'app/components/modal/__close/modal__close';

import {closeConfigPreview} from 'app/actions/config-preview';
import {fetchSchema} from 'app/actions/schema';
import {getConfigPreview} from 'app/reducers/index';
import {getService} from 'app/reducers/index';
import {getSchema} from 'app/reducers/service';

import {configType} from 'app/types';
import {ACTIONS} from 'app/enums/actions';

import {getUser} from 'app/lib/user';
import {getPermissions} from 'app/lib/permissions';
import {antiadbUrl} from 'app/lib/url';
import i18n from 'app/lib/i18n';
import {mergeConfig, getSchemaTypeAndDefVal, clearDefaultValues} from 'app/lib/config-editor';

import './config-preview.css';

const EDITOR = {
    FORM: 'form',
    YAML: 'yaml'
};

class ConfigPreview extends React.Component {
    constructor() {
        super();

        this.state = {
            editor: EDITOR.FORM
        };

        this.onModalClose = this.onModalClose.bind(this);
        this.setEditor = this.setEditor.bind(this);
    }

    componentWillUpdate(nextProps) {
        if (nextProps.configData && this.props.configData && nextProps.configData.label_id !== this.props.configData.label_id && nextProps.configData) {
            this.props.fetchSchema(nextProps.configData.label_id);
        }
    }

    onModalClose() {
        this.props.closeConfigPreview(this.props.id);
    }

    setEditor(e) {
        this.setState({
            editor: e.target.value
        });
    }

    render() {
        const user = getUser();
        const permissions = getPermissions();
        let dataSchema = {},
            data = {},
            dataCurrent = {};

        if (this.props.configData && this.props.schema.loaded) {
            dataSchema = getSchemaTypeAndDefVal(this.props.schema.schema || []);
            dataCurrent = clearDefaultValues(this.props.configData.data, this.props.configData.data_settings, dataSchema.values);
            data = mergeConfig(dataSchema.values, dataCurrent);
        }

        return (
            <Modal
                autoclosable
                onDocKeyDown={this.onModalClose}
                onOutsideClick={this.onModalClose}
                visible={this.props.visible}>
                {this.props.configData && this.props.schema.loaded ?
                    [
                        <Bem
                            key='title'
                            block='modal'
                            elem='title'>
                            {i18n('service-page', 'preview-title')} &#34;{this.props.configData.comment}&#34;
                        </Bem>,
                        <ModalClose
                            key='close'
                            onClick={this.onModalClose} />,
                        <Bem
                            key='radio'
                            block='config-preview'
                            elem='radio'>
                            <RadioButton
                                name='tone'
                                theme='normal'
                                size='s'
                                view='default'
                                tone='default'
                                value={this.state.editor}
                                onChange={this.setEditor} >
                                <RadioButton.Radio value={EDITOR.FORM}>
                                    {i18n('service-page', 'form-editor')}
                                </RadioButton.Radio>
                                <RadioButton.Radio value={EDITOR.YAML}>
                                    {i18n('service-page', 'yaml-editor')}
                                </RadioButton.Radio>
                            </RadioButton>
                        </Bem>,
                        <Bem
                            key='body'
                            block='modal'
                            elem='body'>
                            {this.state.editor === EDITOR.FORM ?
                                <ConfigEditor
                                    data={data}
                                    dataSettings={this.props.configData.data_settings}
                                    dataCurrent={dataCurrent}
                                    schema={this.props.schema.schema}
                                    onError={this.onErrorData}
                                    readOnly /> :
                                <ConfigYamlEditor
                                    data={YAML.safeDump(dataCurrent, {
                                        lineWidth: Infinity
                                    })}
                                    dataSettings={YAML.safeDump(this.props.configData.data_settings, {
                                        lineWidth: Infinity
                                    })}
                                    readOnly />}
                        </Bem>,
                        <Bem
                            key='actions'
                            block='modal'
                            elem='actions'>
                            {user.hasPermission(permissions.CONFIG_CREATE) && (
                                <Link
                                    to={antiadbUrl(`/service/${this.props.serviceId}/label/${this.props.configData.label_id}/${ACTIONS.CONFIGS}/${this.props.id}`)}>
                                    <Button
                                        theme='link'
                                        view='default'
                                        tone='grey'
                                        size='s'
                                        mix={{
                                            block: 'modal',
                                            elem: 'action'
                                        }}
                                        onClick={this.onModalClose}>
                                        {i18n('common', 'duplicate')}
                                    </Button>
                                </Link>
                            )}
                            <Button
                                theme='normal'
                                view='default'
                                tone='grey'
                                size='s'
                                mix={{
                                    block: 'modal',
                                    elem: 'action'
                                }}
                                onClick={this.onModalClose}>
                                {i18n('common', 'close')}
                            </Button>
                        </Bem>
                    ] :
                    ''
                }
            </Modal>
        );
    }
}

ConfigPreview.propTypes = {
    id: PropTypes.number,
    visible: PropTypes.bool,
    serviceId: PropTypes.string,
    configData: configType,
    schema: PropTypes.object,
    closeConfigPreview: PropTypes.func.isRequired,
    fetchSchema: PropTypes.func.isRequired
};

export default connect(state => {
    return {
        id: getConfigPreview(state).id,
        serviceId: getConfigPreview(state).serviceId,
        visible: getConfigPreview(state).visible,
        schema: getSchema(getService(state)),
        configData: getConfigPreview(state).configData
    };
}, dispatch => {
    return {
        closeConfigPreview: id => {
            dispatch(closeConfigPreview(id));
        },
        fetchSchema: labelId => {
            return dispatch(fetchSchema(labelId));
        }
    };
})(ConfigPreview);
