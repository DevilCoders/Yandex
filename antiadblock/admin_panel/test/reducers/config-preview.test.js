import reducer from 'app/reducers/config-preview';
import * as configPreviewActionFixtures from '../fixtures/actions/config-preview';
import {CONFIG_PREVIEW_EMPTY} from '../fixtures/store';

describe('config-preview reducer', () => {
    const initialState = CONFIG_PREVIEW_EMPTY;

    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle OPEN_CONFIG_PREVIEW', () => {
        expect(
            reducer(initialState, configPreviewActionFixtures.OPEN_CONFIG_PREVIEW)
        ).toMatchSnapshot();
    });

    it('should handle CLOSE_CONFIG_PREVIEW', () => {
        const afterOpen = reducer(initialState, configPreviewActionFixtures.OPEN_CONFIG_PREVIEW);

        expect(
            reducer(afterOpen, configPreviewActionFixtures.CLOSE_CONFIG_PREVIEW)
        ).toMatchSnapshot();
    });
});
