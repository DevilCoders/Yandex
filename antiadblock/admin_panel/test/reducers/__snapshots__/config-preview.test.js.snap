// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-preview reducer should handle CLOSE_CONFIG_PREVIEW 1`] = `
Object {
  "configData": null,
  "id": null,
  "serviceId": null,
  "visible": false,
}
`;

exports[`config-preview reducer should handle OPEN_CONFIG_PREVIEW 1`] = `
Object {
  "configData": Object {
    "comment": "made a config",
    "data": Object {
      "field1": "value1",
      "field2": "value2",
      "field3": "value3",
    },
    "date": "2018-01-10T11:12:13",
    "id": 234,
    "statuses": Array [
      Object {
        "comment": "",
        "status": "active",
      },
    ],
  },
  "id": 234,
  "serviceId": 123,
  "visible": true,
}
`;

exports[`config-preview reducer should return the initial state 1`] = `
Object {
  "configData": null,
  "id": null,
  "serviceId": null,
  "visible": false,
}
`;
